quipucords/scanner/network/utils.py

```diff
@@ -173,3 +182,60 @@ def get_fact_names():
 def raw_facts_template():
     """Results template for fact collection on network scans."""
     return {fact_name: None for fact_name in get_fact_names()}
+
+
+GEN_KEYFILES_DIRECTORY = "quipucords_generated_ssh_private_keys"
+GEN_KEYFILE_PREFIX = "quipucords_ssh_private_key_credential"
+
+
+def generate_ssh_keyfile(credential, ssh_key):
+    """Generate an ssh private keyfile for the ssh key content specified."""
+    ssh_pkey_dir = Path(tempfile.gettempdir()) / GEN_KEYFILES_DIRECTORY
+    ssh_pkey_dir.mkdir(parents=True, exist_ok=True)
+    private_keyfile_path = tempfile.NamedTemporaryFile(
+        prefix=f"{GEN_KEYFILE_PREFIX}_{credential.get('id')}_", dir=ssh_pkey_dir
+    ).name
+    with open(private_keyfile_path, "w+") as private_key_fp:
+        private_key_fp.write(ssh_key)
+        private_key_fp.write("\n")
```

Comment:
w8... so the key has 2 line breaks in the end? (the one preserved on the serializer + this)

Reply:
I had that NL added early on because the SSH key file MUST be NL terminated or it will fail to authenticate, so in the cases where someone adds/edits one without (API or through QPC), it will guarantee it working and it does not hurt to have the extra NL.